runtime/service.go

```diff
@@ -315,15 +315,16 @@ func (s *service) StartShim(shimCtx context.Context, containerID, containerdBina
 		}
 
 		s.vmID = uuid.String()
+		s.logger = s.logger.WithField("vmID", s.vmID)
 
 		// If the client didn't specify a VMID, this is a single-task VM and should thus exit after this
 		// task is deleted
 		containerCount = 1
 		exitAfterAllTasksDeleted = true
 
-		log.Infof("will start a single-task VM %s since no VMID has been provided", s.vmID)
+		log.Info("will start a single-task VM since no VMID has been provided")
```

Comment:
I think this and below should be `s.logger`, right?

Reply:
Ah wait, after re-reading this I believe this line should actually stay as `log.Info` but you need to change line 318 to from `s.logger = s.logger.WithField(...)` to `log = log.WithField(...)`. 

`s.logger` is the shim's overall logger. However, this method `StartShim` actually only runs in a short lived process invoked by containerd to create the real, long-running shim process, so setting `s.logger` here won't have any effect (confusing, I know, [the containerd docs have more on this](https://github.com/containerd/containerd/blob/master/runtime/v2/README.md#start))